luminance-glutin/src/lib.rs

```diff
@@ -138,13 +129,27 @@ impl GlutinSurface {
     samples: u16,
   ) -> Result<(Self, EventLoop<()>), GlutinError> {
     let event_loop = EventLoop::new();
-
     let windowed_ctx = ContextBuilder::new()
       .with_gl(GlRequest::Specific(Api::OpenGl, (3, 3)))
       .with_gl_profile(GlProfile::Core)
       .with_multisampling(samples)
-      .with_double_buffer(Some(true))
-      .build_windowed(window_builder, &event_loop)?;
+      .with_double_buffer(Some(true));
+    let surface = Self::new_gl33_with_builders(&event_loop, window_builder, windowed_ctx)?;
+
+    Ok((surface, event_loop))
+  }
+
+  /// Create a new [`GlutinSurface`] by passing in the builders.
+  ///
+  /// This is the most flexible but least hand-holding way of creating a [`GlutinSurface`].
+  ///
+  /// The OpenGL context should be version 3.3, core profile, and double-buffered.
+  pub fn new_gl33_with_builders<T>(
+    event_loop: &EventLoop<T>,
+    window_builder: WindowBuilder,
+    context_builder: ContextBuilder<NotCurrent>,
+  ) -> Result<Self, GlutinError> {
+    let windowed_ctx = context_builder.build_windowed(window_builder, &event_loop)?;
 
     let ctx = unsafe { windowed_ctx.make_current().map_err(|(_, e)| e)? };
```

Comment:
The new function sets the context here. Not sure what you mean in your comment.

Reply:
I mean, the OpenGL 3.3 context. Here if the cliente doesn’t call the right function `.with_gl` etc.), it will not work as intended.

Reply:
Well, if you'd like, you can override the code's choice by calling `.with_gl(GlRequest::Specific(Api::OpenGl, (3, 3))).with_gl_profile(GlProfile::Core)` on the passed-in builder, but as it stands, it's impossible to set other useful options on the context, like vsync or srgb, or use an event loop with a generic type, because luminance needlessly takes control over those parts of the code.

Reply:
Yes, that should be done (and documented). There is a similar function in the `luminance-glfw` crate, [here](https://github.com/phaazon/luminance-rs/blob/master/luminance-glfw/src/lib.rs#L85-L88); even though you can access to the context, you can see that the crate still setup the context correctly [here](https://github.com/phaazon/luminance-rs/blob/master/luminance-glfw/src/lib.rs#L97-L103)

Reply:
So [an example of creating a GLFW window](https://github.com/phaazon/luminance-rs/blob/master/examples/desktop/src/main.rs#L100-L111), people don’t have to think about it.

Reply:
I've updated the PR to set the context version and profile explicitly, and rebased to master.